### ISSUE TITLE
Add endpoint to list job versions

### ISSUE THREAD

**Issue by wslulciuc** [MarquezProject/marquez#614]
Let's add an endpoint that allows the caller to list version metadata for a given job. For example:

## `GET` /namespaces/`{namespace}`/jobs/`{job}`/versions

```
{
  "versions": [
    { ... } /* An array of job objects */
  ]
}
```

We'll also want to return the input and output datasets of the job version.

**Comment by julienledem** [comment]
I think the array should contain Job objects with the same schema as the /jobs endpoint. We should add pagination

**Comment by wslulciuc** [comment]
@julienledem Yep, updated the payload. But, I think we'll also want to add a `version` property to the job object.

**Comment by julienledem** [comment]
WIP: https://github.com/MarquezProject/marquez/tree/feature/job-version-api

**Comment by wslulciuc** [comment]
Fixed #1462
